src/adaptors/tcp/tcp_adaptor.c

```diff
@@ -1089,6 +1115,21 @@ static void handle_connection_event(pn_event_t *e, qd_server_t *qd_server, void
     case PN_RAW_CONNECTION_WAKE: {
         qd_log(LOG_TCP_ADAPTOR, QD_LOG_DEBUG, "[C%" PRIu64 "] PN_RAW_CONNECTION_WAKE %s", conn->conn_id,
                qdr_tcp_connection_role_name(conn));
+        if (CLEAR_ATOMIC_FLAG(&conn->check_idle_conn)) {
```

Comment:
Can we do this bit of code only if conn->ingress is true ?

Reply:
Good idea! How about we check for existence of the idle_timer? The patch already does that elsewhere since the idle_timer is only allocated if idleTimeoutSeconds != 0.

Reply:
Cool yo